src/providers/postgres/qgspostgresutils.cpp

```diff
@@ -658,3 +658,53 @@ QString QgsPostgresUtils::variantMapToHtml( const QVariantMap &variantMap, const
   }
   return result;
 }
+
+bool QgsPostgresUtils::setProjectComment( QgsPostgresConn *conn, const QString &projectName, const QString &schemaName, const QString &comment )
+{
+  if ( !projectsTableExists( conn, schemaName ) )
+    return false;
+
+  if ( !columnExists( conn, schemaName, QStringLiteral( "qgis_projects" ), QStringLiteral( "comment" ) ) )
+  {
+    if ( !addCommentColumnToProjectsTable( conn, schemaName ) )
+      return false;
+  }
+
+  const QString sql = QStringLiteral( "UPDATE %1.qgis_projects SET comment = %3 WHERE name = %2" )
+                        .arg( QgsPostgresConn::quotedIdentifier( schemaName ), QgsPostgresConn::quotedValue( projectName ), QgsPostgresConn::quotedValue( comment ) );
+
+  QgsPostgresResult res( conn->PQexec( sql ) );
+  return res.PQresultStatus() == PGRES_COMMAND_OK;
+}
+
+QString QgsPostgresUtils::projectComment( QgsPostgresConn *conn, const QString &schemaName, const QString &projectName )
+{
+  if ( !projectsTableExists( conn, schemaName ) || !columnExists( conn, schemaName, QStringLiteral( "qgis_projects" ), QStringLiteral( "comment" ) ) )
+    return QString();
```

Comment:
Again, can't we skip the checks? The select query will just fail if table or column is missing.